### DISCUSSION TITLE
String and function esthetics

### DISCUSSION THREAD

**Discussion by jean-claudeF** [micropython/micropython#16375] [Using MicroPython]
I love the clear structure of Python.
One thing however hurts my feelings.

Example:

```
def test():
    s = """blahblah
blahblah
blahblah
"""
    print(s)
```

This is not beautiful.
For me the whole string definition should be indented.
Is  there a way to do it more elegantly?

**Comment by robert-hh** [comment]
Only in a different way clumsy:
```
def test():
    s = "blahblah\n"\
        "blahblah\n"\
        "blahblah"
    print(s)
```

**Comment by andrewleech** [comment]
Yeah I personally agree it's pretty ugly like this. 

Often I define my multiline strings at global context just to avoid this. 

Alternatively:
```
def test():
    s = (
        "blahblah\n"
        "blahblah\n"
        "blahblah\n"
    )
    print(s)
```

**Comment by corranwebster** [comment]
In CPython you'd typically use the standard library `textwrap.dedent` for this sort of thing, so I was going to suggest mip-installing it and doing:
``` python
def test():
    s = dedent("""\
        blahblah
        blahblah
        blahblah
        """)
    print(s)
```
assuming you were willing to pay the runtime cost.

But apparently `textwrap` in micropython-lib has a bug: https://github.com/micropython/micropython-lib/issues/318

That said, if it's constant do what @andrewleech suggested and make it a module-level string.

**Comment by jean-claudeF** [comment]
I use global variables now at module level, that seems to me the most elegant solution.
Like this:

```
CMD_LISTFILES = """import os
files = os.listdir()
for f in files:
    print(f)
"""
```
